staticd/static_nb_config.c

```diff
@@ -1122,6 +1170,44 @@ int routing_control_plane_protocols_control_plane_protocol_staticd_route_list_sr
 	return NB_OK;
 }
 
+/*
+ * XPath:
+ * /frr-routing:routing/control-plane-protocols/control-plane-protocol/frr-staticd:staticd/route-list/src-list/path-list/frr-nexthops/nexthop/srte-color
+ */
+int routing_control_plane_protocols_control_plane_protocol_staticd_route_list_src_list_path_list_frr_nexthops_nexthop_color_modify(
+	struct nb_cb_modify_args *args)
+{
+	switch (args->event) {
+	case NB_EV_VALIDATE:
+	case NB_EV_PREPARE:
+	case NB_EV_ABORT:
+		break;
+	case NB_EV_APPLY:
+		if (static_nexthop_color_modify(args) != NB_OK)
+			return NB_ERR;
+
+		break;
+	}
+	return NB_OK;
+}
+
+
+int routing_control_plane_protocols_control_plane_protocol_staticd_route_list_src_list_path_list_frr_nexthops_nexthop_color_destroy(
+	struct nb_cb_destroy_args *args)
+{
+	/* srte-color is a uint32 type with default value,
+	 * so no need to do any operations in destroy callback
+	 */
```

Comment:
The nexthop color needs to be unset here manually, regardless if it has a default value or not (because of the "when" statement).

An an example, if the nexthop type changes from `ip4` to `ifindex`, this leaf will be deleted automatically, which will trigger this callback. The nexthop color needs then to be unset in the corresponding internal data structure.